pkg/cmd/root/root.go

```diff
@@ -134,3 +125,30 @@ func bareHTTPClient(f *cmdutil.Factory, version string) func() (*http.Client, er
 		return factory.NewHTTPClient(f.IOStreams, cfg, version, false)
 	}
 }
+
+func newCodespaceCmd(f *cmdutil.Factory) *cobra.Command {
+	cmd := ghcs.NewRootCmd(ghcs.NewApp(
+		output.NewLogger(f.IOStreams.Out, f.IOStreams.ErrOut, !f.IOStreams.IsStdoutTTY()),
+		ghcsApi.New("", &lazyLoadedHTTPClient{factory: f}),
+	))
+	cmd.Use = "codespace"
+	cmd.Aliases = []string{"cs"}
+	cmd.Hidden = true
+	return cmd
+}
+
+type lazyLoadedHTTPClient struct {
+	factory    *cmdutil.Factory
+	httpClient *http.Client
+}
+
+func (l *lazyLoadedHTTPClient) Do(req *http.Request) (*http.Response, error) {
+	if l.httpClient == nil {
+		var err error
+		l.httpClient, err = l.factory.HttpClient()
+		if err != nil {
+			return nil, err
+		}
+	}
+	return l.httpClient.Do(req)
+}
```

Comment:
@mislav do we need a `sync.RWMutex` here?

Reply:
Good point! Please add one 🙏

Reply:
Done https://github.com/cli/cli/pull/4384/commits/c3ce95ea1c140297122812b3aa87a831cc291ff0

Do you think the lazy instantiation is worth it? Most if not all codespace-related commands require a network call, so I am curious why you went the lazy route in this case.